src/main/java/com/netuitive/dropwizard/metrics/NetuitiveMetricsCloudReporter.java

```diff
@@ -0,0 +1,323 @@
+package com.netuitive.dropwizard.metrics;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+import org.apache.commons.lang3.math.NumberUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Metered;
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.ScheduledReporter;
+import com.codahale.metrics.Snapshot;
+import com.codahale.metrics.Timer;
+import com.netuitive.iris.client.AbstractRestClient;
+import com.netuitive.iris.client.metric.NetuitiveIngestMetricRestClient;
+import com.netuitive.iris.client.request.metric.IngestRequest;
+import com.netuitive.iris.entity.Element;
+import com.netuitive.iris.entity.Metric;
+import com.netuitive.iris.entity.Sample;
+
+public class NetuitiveMetricsCloudReporter extends ScheduledReporter  {
+
+    private static final Logger LOG = LoggerFactory.getLogger(NetuitiveMetricsCloudReporter.class);
+
+    private final String apiHost;
+    private final String apiKey;
+    private final String elementName;
+    private final String elementType;
+    private NetuitiveIngestMetricRestClient client;
+    
+    private NetuitiveMetricsCloudReporter(MetricRegistry registry,
+                            String apiKey, 
+                            String apiHost, 
+                            String elementName, 
+                            String elementType, 
+                            TimeUnit rateUnit,
+                            TimeUnit durationUnit,
+                            MetricFilter filter) {
+        super(registry, "netuitive-cloud-reporter", filter, rateUnit, durationUnit);
+        this.apiKey = apiKey;
+        this.apiHost = (apiHost == null || apiHost.length() == 0 ? AbstractRestClient.API_HOST : apiHost);
+        this.elementName = elementName;
+        this.elementType = (elementType == null || elementType.length() == 0 ? "Dropwizard" : elementType);
+
+        client = new NetuitiveIngestMetricRestClient(AbstractRestClient.SCHEME, this.apiHost, this.apiKey);
+    }
+
+    /**
+     * Returns a new {@link Builder} for {@link NetuitiveMetricsCloudReporter}.
+     *
+     * @param registry the registry to report
+     * @return a {@link Builder} instance for a {@link NetuitiveMetricsCloudReporter}
+     */
+    public static Builder forRegistry(final MetricRegistry registry) {
+        return new Builder(registry);
+    }
+
+    /**
+     * A builder for {@link NetuitiveMetricsCloudReporter} instances.
+     */
+    @NotThreadSafe
+    public static final class Builder {
```

Comment:
I recommend using the Lombok `@Builder` annotation for this, it can shrink this class significantly.